source/main.cc

```diff
@@ -550,13 +551,36 @@ template<int dim>
 void
 run_simulator(const std::string &input_as_string,
               const bool output_xml,
-              const bool output_plugin_graph)
+              const bool output_plugin_graph,
+              const bool validate_only)
 {
   using namespace dealii;
 
   ParameterHandler prm;
   const bool i_am_proc_0 = (Utilities::MPI::this_mpi_process(MPI_COMM_WORLD) == 0);
   aspect::Simulator<dim>::declare_parameters(prm);
+
+  if (validate_only)
+    {
+      try
+        {
+          parse_parameters (input_as_string, prm);
+        }
+      catch (...)
+        {
+          throw aspect::QuietException();
+        }
+      if (i_am_proc_0)
+        std::cerr << "The provided parameter file is valid.\n\n"
```

Comment:
`std::cerr` is not the correct place: that's where errors go, but you print a success message. Can you still change this?